### DISCUSSION TITLE
Variant pairing question

### DISCUSSION THREAD

**Discussion by trudator** [ORTools4MTEX/ORTools#26] [Q&A] [answered]
Hi Frank, 

Just a quick question regarding your comment in our [last discussion](https://github.com/ORTools4MTEX/ORTools/discussions/19): `Variants 7 to 24 are just repetitions of the same variants within different packets, but share the same characteristics as variants 1 to 6. The equivalent variants of variant 1 for example are 1, 7, 13 and 19.` Could you explain this statement a little bit or share any paper saying the similarities in pairing here? 

I saw some authors also did variants pairing outside of a packet (e.g., V1-V16). Would we be able to do that with `plotMap_variantPairs`?

Thanks Frank and have a nice day.
Linh

**Comment by frankNiessen** [comment, marked as answer]
Hi Linh, 

have a look at Table 2 in the paper 

<div class="csl-entry">Morito, S., Tanaka, H., Konishi, R., Furuhara, T., &#38; Maki, T. (2003). The morphology and crystallography of lath martensite in Fe-C alloys. <i>Acta Materialia</i>, <i>51</i>, 1789–1799. https://doi.org/10.1016/j.actamat.2006.07.009</div>


Here you have listed all 24 variants of the Kurdjumov-Sachs (K-S) orientation relationship. You will see that groups of 6 variants are assigned to a packet (1-6, 7-12, 13-18 and 19-24). A packet is the collection of variants that form on a specific habit plane. Within each packet there are 6 different combinations of how the parallel crystal directions can be paired. This is illustrated in Figure 1 of the paper. The misorientations between the variants within one packet are the same for each group of 6 variants. 

Let me demonstrate this: Considering you have defined the K-S orientation relationship as shown in [Example 1](https://github.com/ORTools4MTEX/ORTools/blob/develop/ORTools_example1.m), you can check the misorientation of the first 6 variants with the 1st variant:

`KS.variants(1:6).*inv(KS.variants(1))`
```
  Bunge Euler angles in degree
     phi1     Phi    phi2
        0       0       0
      315 109.471     225
  140.768 75.5225 320.768
  86.2723  172.56 93.7277
  219.232 75.5225 39.2315
  33.0855 100.084 146.915
```
Doing the same for variants 7 to 12 with respect to variant 7 gives the same misorientations:

`KS.variants(7:12).*inv(KS.variants(7))`
```
  Bunge Euler angles in degree
     phi1     Phi    phi2
        0       0       0
      315 109.471     225
  140.768 75.5225 320.768
  86.2723  172.56 93.7277
  219.232 75.5225 39.2315
  33.0855 100.084 146.915
```
This means that the boundaries that the variants form within each packet are equivalent, they "share the same characteristics". Therefore we translate all 24 variants in [plotMap_variantPairs](https://github.com/ORTools4MTEX/ORTools/blob/develop/src/plotMap_variantPairs.m) to their equivalent variants 1 to 6 and check the fraction of grain boundaries between the different variants. This type of analysis is described here:

<div class="csl-entry">Morito, S., Pham, A. H., Hayashi, T., &#38; Ohba, T. (2015). Block boundary analyses to identify martensite and bainite. <i>Materials Today: Proceedings</i>, <i>2</i>, S913–S916. https://doi.If you really want to do this, you could filter your dataset to only get variants of a certain packet and limit the naorg/10.1016/j.matpr.2015.07.430</div>

I cannot see the point in doing a variant pairing analysis for all 24 variants, it does not give you any additional information. 

Best wishes
Frank

**Reply by trudator** [reply]
Hi Frank,

Thanks for such a thorough answer!
Just a quick follow-up question: why do you consider V1-V3 and V1-V5 the same while their misorientations relative to V1 is quite different (based on your above tables)?

`I cannot see the point in doing a variant pairing analysis for all 24 variants, it does not give you any additional information.` It just triggers out of my curiosity while reading some papers saying that a high fraction of V1-16 was observed in high-C steel (https://doi.org/10.1016/j.actamat.2012.09.046).

Anyway, thanks so much for your answer!

**Reply by frankNiessen** [reply]
They are not, V1 to V6 are unique, and V7 to V12 for instance are equivalent to V1 to V6, just on a different habit plane.

Best wishes
Frank

**Reply by trudator** [reply]
Hi, I'm asking about V1-V3(V5) as marked in the below image. Why is it considered as similar?

![image](https://user-images.githubusercontent.com/117386810/206762702-0dc46f6c-1609-4ffe-9566-d8051c58ab3d.png)

**Comment by frankNiessen** [comment]
Because the boundary misorientations are the same, 60 degrees around [011] as stated in the above referenced paper. 

![Screenshot_20221209_195125_Office](https://user-images.githubusercontent.com/41304940/206773504-67e3da68-cb08-4b7c-ba2d-359448ee0610.jpg)

**Reply by trudator** [reply]
Thanks Frank!

**Comment by trudator** [comment]
Hi Frank, 

I realized that there are some other pairings with characteristics similar to the typical pairing with V1, the same as what Morito et al. described in Table 3. For example, V2/V5 & V3/V6 are similar to V1/V4, or V3/V2 & V5/V4 are similar to V1-V6. In the `plotMap_variantPairs`, do you consider (sum them all together) them too? If not, do you think those V1-coupled pairings could represent them all?

![image](https://user-images.githubusercontent.com/117386810/208050768-3a7b3d70-bd0d-422c-a1ca-851886e9efbc.png)

Morito et al. paper: Morito, S., Tanaka, H., Konishi, R., Furuhara, T., & Maki, T. (2003). The morphology and crystallography of lath martensite in Fe-C alloys. Acta Materialia, 51, 1789–1799. https://doi.org/10.1016/j.actamat.2006.07.009

Thanks Frank.
Linh.

**Reply by frankNiessen** [reply]
Hi Linh, 

as you can see in the table, they consider three groups of variant pairs which are associated with three characteristic misorientations. For some reasons they do not consider the V1-V4 pairing in this list, but the Morito reference I gave earlier shows that this one also has a characteristic misorientation. 

The point is that it does not matter whether you pair V4-V6 or V3-V5, as the characteristic of the grain boundary, which is defined by the misorientation, is the same. From a technological point of view it is only important to know the character of the boundaries that define the blocks, as this will give you the properties of the martensitic structure. When we plot the boundaries of, say V1-V3 in our ORTools function, we also check for the equivalent pairs of V3-V5, V5-V1 etc and include them in the V1-V3-type boundaries. 

Best wishes
Frank

**Reply by trudator** [reply]
Thanks Frank for the explanation!

**Comment by wanghaijian217** [comment]
It's a great honor to communicate with you. If possible, can you give the statistics of the boundary length ratio of the remaining variant pairs (16 kinds, Figure 5 in DOI: 10.1016/j.actamat.2011. 12.018)? I would appreciate it if you could make corresponding codes.

**Reply by frankNiessen** [reply]
Hi, 

this would in fact be interesting, I am currently not having the time unfortunately. I started a new conversation #39 in which you can list this feature for implementation in the future, thanks!

Best wishes
Frank

**Reply by Zhangcl-2024** [reply]
Dear Mr. Wang,

I hope this message finds you well. Sorry to bother you. I would like to ask whether you have already generated the densities for the remaining variant pairs, namely:
“V1-V7,” “V1-V8,” “V1-V9 (V19),” “V1-V10 (V14),” “V1-V11 (V13),”
“V1-V12 (V20),” “V1-V15 (V23),” “V1-V16,” “V1-V17,” “V1-V18 (V22),”
“V1-V21,” and “V1-V24.”

Currently, the [plotMap_variantPairs] function only provides densities for “V1-V2,” “V1-V3 (V5),” “V1-V6,” and “V1-V4.” If possible, could you kindly provide the densities for these remaining variant pairs?

Thank you very much for your time and assistance.
![233360980-5494753f-067e-4468-a5e8-0e474c0da059](https://github.com/user-attachments/assets/6826acfa-d995-4fb4-a005-5c9bf943a652)

**Comment by Zhangcl-2024** [comment]
Hi Frank,

I sincerely appreciate the efforts you and your colleagues have invested here. Through your discussions, I have recently developed a strong interest in ORTools and have learned a great deal about variants and variant pairs.

In the plotMap_variantPairs code, only the variant pairs “V1-V2,” “V1-V3 (V5),” “V1-V6,” and “V1-V4” are presented. If possible, could you kindly provide the densities for the remaining variant pairs as well, namely:

“V1-V7,” “V1-V8,” “V1-V9 (V19),” “V1-V10 (V14),” “V1-V11 (V13),”
“V1-V12 (V20),” “V1-V15 (V23),” “V1-V16,” “V1-V17,” “V1-V18 (V22),”
“V1-V21,” and “V1-V24.”

Since the densities of these remaining variant pairs are also important, I aim to incorporate all 16 variant pairs to better analyze their combined impact on impact toughness.

Thank you very much for your consideration.
![233360980-5494753f-067e-4468-a5e8-0e474c0da059](https://github.com/user-attachments/assets/83291bc0-2210-4109-b606-3da9f7ff95e7)
